frontend/src/components/GoogleMapLoader.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <div>
+    <div class="google-map" ref="googleMap"></div>
+    <template v-if="Boolean(map)">
+      <slot :map="map" />
+    </template>
+  </div>
+</template>
+
+<script lang="ts">
+/* eslint-disable no-undef */
+import { defineComponent, PropType } from "vue";
+import { Loader } from "@googlemaps/js-api-loader";
+
+export default defineComponent({
+  props: {
+    mapConfig: {
+      type: Object as PropType<google.maps.MapOptions>,
+      required: true,
+    },
+    apiKey: {
+      type: String,
+      default: "",
+    },
+    markersOptions: {
+      type: Array as PropType<google.maps.MarkerOptions[]>,
+      default: () => [],
+    },
+  },
+
+  data() {
+    return {
+      map: null as google.maps.Map | null,
```

Comment:
for my learning, what does this mean?

Reply:
hmm, may not be super precise here, but I think what this line does is that is allows you to do something like `this.map` elsewhere in this file. I think of it as an attribute of a class and I initialised it to null. `google.maps.Map | null` just means the type is nullable